Update for elm 0.18

dillonkearns opened the pull request.

dillonkearns committed:
Update for elm 0.0.18

dillonkearns committed:
Add package.json with test-watch script

pierrebeitz commented on src/GildedRose.elm:
```diff
@@ -16,7 +16,7 @@ updateQuality =
 updateQualityItem (Item name sellIn quality) =
     let
-        quality' =
+        quality_ =
```
as the reason for deprecating `'`s was to force better names i guess we should aim for something like `updatedQuality`. What do you think?

dillonkearns replied:
That's definitely a good practice, although this is a legacy code refactoring exercise so having unclear names could be good? I don't have a strong opinion, could go either way.

pierrebeitz replied:
oh! good point. so be it!

pierrebeitz commented on elm-package.json:
```diff
@@ -8,9 +8,9 @@
-        "elm-lang/html": "1.1.0 <= v < 2.0.0"
+        "elm-community/elm-test": "3.1.0 <= v < 4.0.0",
+        "elm-lang/core": "5.1.1 <= v < 6.0.0",
+        "elm-lang/html": "2.0.0 <= v < 3.0.0"
```
Do we actually need `elm-lang/html`. I think i put it in here by mistake. Could you please verify?

dillonkearns replied:
That's true, and elm-test isn't needed here either.

pierrebeitz commented:
thank you so much, @dillonkearns!

pierrebeitz reviewed (commented):
i'm fine with introducing yarn. but please update the readme accordingly!

dillonkearns committed:
Remove unused dependencies

dillonkearns committed:
Remove yarn lock file

dillonkearns commented:
@pierrebeitz sorry for the delay, finally addressed your comments. As for yarn test, I removed the lock file for now to avoid confusion, so people can choose to use it or not and it shouldn't be an issue.

pierrebeitz commented:
looks good to me! thanks a lot!

dillonkearns commented:
👍 neat, and thanks for sharing the exercise, @pierrebeitz!
